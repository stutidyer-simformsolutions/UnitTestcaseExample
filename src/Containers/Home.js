import React from "react";
import { Button, StyleSheet, Text, View } from "react-native";

const Home = (props) => {
  return (
    <View style={styles.container}>
      <Text
        style={styles.textStyle}
      >{`Hello Everyone! \n Welcome to Unit Testcase Demo!`}</Text>
      <View style={styles.buttonContainer}>
        <Button
          title="Demo with Statefull Component"
          onPress={() => props.navigation.navigate("DisplayDataWithClass")}
          style={styles.buttonStyle}
          testID="class-component-button"
        />
        <Button
          title="Demo with Stateless Component"
          onPress={() => props.navigation.navigate("DisplayDataWithFC")}
          style={styles.buttonStyle}
          accessibilityLabel="Press Me"
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  buttonContainer: { flex: 0.8, marginVertical: 25, justifyContent: "center" },
  textStyle: {
    textAlign: "center",
    marginVertical: 8,
    fontSize: 25,
    lineHeight: 50,
  },
  buttonStyle: {
    // height: 120,
    // width: 180
  },
});

export default Home;
