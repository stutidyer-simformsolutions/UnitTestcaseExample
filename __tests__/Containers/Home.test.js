import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import Home from "../../src/Containers/Home";
describe("Home Screen Testcase", () => {
  const navigation = {
    navigate: jest.fn(),
  };

  test("Snapshot of home screen", async () => {
    const { toJSON } = render(<Home navigation={navigation} />);
    expect(await toJSON()).toMatchSnapshot();
  });

  it("Should have title", () => {
    const title = "Hello Everyone! \n Welcome to Unit Testcase Demo!"
    const { getByText } = render(<Home navigation={navigation} />);
    const foundButtonTitle = getByText(title);
    expect(foundButtonTitle.props.children).toEqual(title);
  });

  it("Should have a button and have been called", async () => {
    const { getByTestId } = render(<Home navigation={navigation} />);

    const button = getByTestId("class-component-button");
    fireEvent.press(button);
    expect(await button).toBeTruthy();
  });

  it("Should find the button via accessibilityLabel and have been called", async () => {
    const { getByLabelText } = render(<Home navigation={navigation} />);

    const foundButton = getByLabelText("Press Me");
    fireEvent.press(foundButton);
    expect(await foundButton).toBeTruthy();
  });
});
