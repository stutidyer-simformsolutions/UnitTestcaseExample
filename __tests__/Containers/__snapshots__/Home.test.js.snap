// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Testcase Snapshot of home screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "lineHeight": 50,
        "marginVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Hello Everyone! 
 Welcome to Unit Testcase Demo!
  </Text>
  <View
    style={
      Object {
        "flex": 0.8,
        "justifyContent": "center",
        "marginVertical": 25,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="class-component-button"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Demo with Statefull Component
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Press Me"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Demo with Stateless Component
        </Text>
      </View>
    </View>
  </View>
</View>
`;
